AstSemantics.md

```diff
@@ -111,6 +112,9 @@ Integer loads can specify a *storage size* which is smaller than the result type
 well as a signedness which determines whether the bytes are sign- or zero-
 extended into the result type.
 
+Each `load` and `store` specifies the [linear memory index](Modules.md#index-spaces)
+to operate on with an unsigned integer immediate.
+
```

Comment:
Given that wasm is designed to allow multiple instances to coexist in the same process, there is already a "segmented" memory model: this PR simply removes the restriction that a single instance can only access 1 memory.  By default, the expectation is that all wasm modules would use a single memory.  From a C++ pov, multiple memories would correspond to the (also statically numbered) [address space](http://llvm.org/docs/LangRef.html#pointer-type) in LLVM (and address spaces would provide a direct way to access this functionality from C++).

I agree it's not strictly necessary in the MVP, but there is a distinct symmetry with tables and I think it's something we definitely want in the long-term, so at the very least, we should generalize the design so that N heaps is possible but constrain to 0..1 (like we've done in other cases).  But if we do multiple tables anyway and make the refactorings to allow better code sharing and less code-dirtying on cache hit (viz., getting away from patching), then I think multiple heaps isn't actually much work in practice.